Task8_0.cpp

```diff
@@ -0,0 +1,104 @@
+// Гурьянов Илья, гр. 1/147.
+// Задание №8, номер 0 (животные)
+
+#include <iostream>
+#include <string>
+
+using namespace std;
+
+class Animal
+{
+protected:
+    string name;
+    string color;
+    int age;
+public:
+    Animal(string n, int a, string c): name(n), color(c), age(a) {}
+    string getName() {return name;}
+    virtual void Show() = 0;
+    virtual string Speak() = 0;
+};
+
+
+class Dog: public Animal
+{
+public:
+    Dog(string name, int age, string color, string poroda): Animal(name, age, color)
+    {
+        this->poroda = poroda;
+    }
+    void Show() override
+    {
+        cout << name << "\t" << age << "\t" << color << "\t" << poroda << endl;
+    }
+    string Speak() override
+    {
+        return "gav gav gav";
+    }
+private:
+    string poroda;
+};
+
+
+class Cat: public Animal
+{
+public:
+    Cat(string name, int age, string color, int year): Animal(name, age, color)
+    {
+        this->year = year;
+    }
+    void Show() override
+    {
+        cout << name << "\t" << age << "\t" << color << "\t" << year << endl;
+    }
+    string Speak() override
+    {
+        return "meow meow meow";
+    }
+private:
+    int year;
+};
+
+
+class Raccoon : public Animal
+{
+    string habitat; // ареал обитания енота
+    float weight; // вес енота
+    float length; // длина енота
+
+public:
+    Raccoon(string name, int age, string color, string habitat, float weight, float length) : Animal(name, age, color)
+    {
+        this->habitat = habitat;
+        this->weight = weight;
+        this->length = length;
+    }
+    void Show() override
+    {
+        cout << "Name: " << name << "; age: " << age << "; color: " << color << "; habitat: " << habitat << "; weight: " << weight << "; length: " << length << endl;
+    }
+    string Speak() override
+    {
+        return "'yeee yee yee'";

```

Comment:
https://youtu.be/3F-Q-r85Bow?t=2

Reply:
Raccoons Noises 
https://www.youtube.com/watch?v=p1OjsFVEnwM&list=LL&index=1

Reply:
> https://youtu.be/3F-Q-r85Bow?t=2
Raccoons Noises
https://youtu.be/p1OjsFVEnwM